Gems/RecastNavigation/Code/Source/Misc/RecastNavigationMeshCommon.h

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <Recast.h>
+#include <AzCore/Component/EntityId.h>
+#include <Misc/RecastHelpers.h>
+#include <Misc/RecastNavigationDebugDraw.h>
+#include <Misc/RecastNavigationMeshConfig.h>
+#include <RecastNavigation/RecastNavigationMeshBus.h>
+
+namespace RecastNavigation
+{
+    //! Common navigation mesh logic for Recast navigation components. Recommended use is as a base class.
+    //! The method provided are not thread-safe. Use the mutex from @m_navObjects to synchronize as necessary at the higher level.
+    class RecastNavigationMeshCommon
+    {
+    public:
+        AZ_RTTI(RecastNavigationMeshCommon, "{D34CD5E0-8C29-4545-8734-9C7A92F03740}");
+        virtual ~RecastNavigationMeshCommon() = default;
+
+        //! Allocates and initializes Recast navigation mesh into @m_navMesh
```

Comment:
uber nit: Missing `.` (repeated in file)

Reply:
Done.